src/propagation.jl

```diff
@@ -1,143 +1,273 @@
+    ss16asteph = TaylorInterpolant(ss16asteph_t0, ss16asteph_t, ss16asteph_x)  # TaylorInterpolant ephemeris
+    # Compute point-mass Newtonian accelerations from ephemeris: all bodies except asteroid
     # accelerations of "everybody else" are needed when evaluating asteroid post-Newtonian acceleration
-    Nm1 = (size(ss16asteph_x)[2]-13) ÷ 6
+    # Number of bodies that contibute to the asteroid's acceleration
+    Nm1 = (size(ss16asteph_x)[2]-13) ÷ 6 
+    # Initialize a TaylorInterpolant for the point-mass Newtonian accelerations
     acc_eph = TaylorInterpolant(ss16asteph.t0, ss16asteph.t, Matrix{eltype(ss16asteph.x)}(undef, length(ss16asteph.t)-1, 3Nm1))
+    # Initialize a TaylorInterpolant for the newtonian N body potential 
     newtonianNb_Potential = TaylorInterpolant(ss16asteph.t0, ss16asteph.t, Matrix{eltype(ss16asteph.x)}(undef, length(ss16asteph.t)-1, Nm1))
+    # Fill TaylorInterpolant.x with zero polynomials
     fill!(acc_eph.x, zero(ss16asteph.x[1]))
     fill!(newtonianNb_Potential.x, zero(ss16asteph.x[1]))
-    _1_to_Nm1 = Base.OneTo(Nm1) # iterator over all bodies except asteroid
+    # Iterator over all bodies except asteroid
+    _1_to_Nm1 = Base.OneTo(Nm1) 
     for j in _1_to_Nm1
         for i in _1_to_Nm1
             # i == j && continue
             if i == j
+                # 
             else
-                X_ij = ss16asteph.x[:,3i-2] .- ss16asteph.x[:,3j-2]
-                Y_ij = ss16asteph.x[:,3i-1] .- ss16asteph.x[:,3j-1]
-                Z_ij = ss16asteph.x[:,3i  ] .- ss16asteph.x[:,3j  ]
+                # Difference between two positions (\mathbf{r}_i - \mathbf{r}_j)
+                X_ij = ss16asteph.x[:,3i-2] .- ss16asteph.x[:,3j-2]  # X-axis component
+                Y_ij = ss16asteph.x[:,3i-1] .- ss16asteph.x[:,3j-1]  # Y-axis component
+                Z_ij = ss16asteph.x[:,3i  ] .- ss16asteph.x[:,3j  ]  # Z-axis component
+                # Distance between two bodies squared ||\mathbf{r}_i - \mathbf{r}_j||^2
                 r_p2_ij = ( (X_ij.^2) .+ (Y_ij.^2) ) .+ (Z_ij.^2)
+                # Distance between two bodies ||\mathbf{r}_i - \mathbf{r}_j||
                 r_ij = sqrt.(r_p2_ij)
+                # Newtonian potential
                 newtonianNb_Potential.x[:,j] .+= (μ[i]./r_ij)
-            end #if i != j
-        end #for, i
-        acc_eph.x[:,3j-2] .= differentiate.(ss16asteph.x[:,3(Nm1+j)-2])
-        acc_eph.x[:,3j-1] .= differentiate.(ss16asteph.x[:,3(Nm1+j)-1])
-        acc_eph.x[:,3j  ] .= differentiate.(ss16asteph.x[:,3(Nm1+j)  ])
-    end #for, j
+            end
+        end
+        # Fill acelerations by differentiating velocities 
+        acc_eph.x[:,3j-2] .= differentiate.(ss16asteph.x[:,3(Nm1+j)-2])  # X-axis component
+        acc_eph.x[:,3j-1] .= differentiate.(ss16asteph.x[:,3(Nm1+j)-1])  # Y-axis component
+        acc_eph.x[:,3j  ] .= differentiate.(ss16asteph.x[:,3(Nm1+j)  ])  # Z-axis component
```

Comment:
Should `differentiate` be used here, or the specialized method defined in PlanetaryEphemeris.jl? (cc: @PerezHz )

Reply:
Sorry for the late reply! I recall correctly, I avoided bumping TaylorSeries to the latest version in `NEOs.jl` precisely due to this issue, so probably it makes sense to use the other method (i.e., the `order`-preserving derivative), but we would have to check

Reply:
I've run `pha/apophis.jl` and the results do not change by using `PlanetaryEphemeris.ordpres_differentiate`.

Reply:
Excellent!